### PR TITLE
Keep file id on move

### PR DESCRIPTION
**Pull request by nickvergessen**
Downstream 27508

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]
looks ok

**Comment by rullzer** [comment]
CI is not happy :disappointed:

**Comment by MorrisJobke** [comment]
> CI is not happy 😞

Fixed by renaming the copied class 🙈

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍 Also fixes #4284

**Comment by MorrisJobke** [comment]
```
PHP Fatal error:  Call to a member function setName() on boolean in /drone/src/github.com/nextcloud/server/3rdparty/sabre/dav/lib/DAV/Tree.php on line 151
```

@rullzer @icewind1991 @nickvergessen Can one of you look into this?

**Comment by MorrisJobke** [comment]
@nickvergessen Unit Tests are still failing

**Comment by MorrisJobke** [comment]
@nickvergessen Backport?

**Comment by nickvergessen** [comment]
Don't really feel like backporting a 800 lines patch like this to 11....
But if all agree I can do it.

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 1]
```diff
@@ -284,6 +286,10 @@ public function handleGetProperties(PropFind $propFind, \Sabre\DAV\INode $node)
 		$httpRequest = $this->server->httpRequest;
 
 		if ($node instanceof \OCA\DAV\Connector\Sabre\Node) {
+			if (!$node->getFileInfo()->isReadable()) {
+				// avoid detecting files through this means
+				throw new NotFound();
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php]: Regression, this breaks dir listing for folders which have an item blocked by files access control:
https://github.com/nextcloud/files_accesscontrol/issues/65

**Pull request merged**